### ISSUE TITLE
nvfp4: add padding to handle tensor parallel, size might not be divisible by 16

### ISSUE THREAD

**Issue by vkuzo** [pytorch/ao#3233]
use case: lm-eval

@andrewor14  add more context

**Comment by andrewor14** [comment]
Sharing my repro so far. Right now vllm is failing for some other reason (can't bind to port) but I think it's related to my local env. Hoping @SalmanMohammadi can add more details but this is how I would reproduce it:

```
import torch
from torchao.prototype.mx_formats import NVFP4InferenceConfig
from transformers import AutoModelForCausalLM, AutoTokenizer, TorchAoConfig

model_id = "Qwen/Qwen3-8B"
quantization_config = TorchAoConfig(quant_type=NVFP4InferenceConfig())
model = AutoModelForCausalLM.from_pretrained(
    model_id,
    device_map="auto",
    dtype=torch.bfloat16,
    quantization_config=quantization_config,
)
tokenizer = AutoTokenizer.from_pretrained(model_id)
save_to = "/tmp/qwen3-8b-nvfp4"
model.save_pretrained(save_to, safe_serialization=False)
tokenizer.save_pretrained(save_to)
```

Eval command:
```
accelerate launch -m lm_eval --model vllm --model_args pretrained="/tmp/qwen3-8b-nvfp4",tensor_parallel_size=8,dtype=auto,gpu_memory_utilization=0.8 --tasks mmlu --batch_size 32
```
